src/utils/value_list/value_list.c

```diff
@@ -416,6 +430,9 @@ plugin_value_list_to_metric_family(value_list_t const *vl, data_set_t const *ds,
     }
     status = status || metric_label_set(&m, name, vl->type_instance);
   }
+  if (strlen(vl->plugin_instance) == 0 && strlen(vl->type_instance) == 0) {
```

Comment:
```suggestion
  else if (strlen(vl->plugin_instance) == 0) {
```

Reply:
The two previous `if` blocks can both be executed. The last block is only executed if *neither* of the first two blocks is executed. Using an `else if` here makes it seem like the last two blocks are more closely related than they actually are, so I prefer to use just an `if` here.

Reply:
On second thought, I think the control flow can be improved and I have done this in 215eee3f791833408a5e43c9b5682481bfbdff93.

Reply:
Much clearer, thanks!